### DISCUSSION TITLE
WebGLRenderer settings Q&A

### DISCUSSION THREAD

**Discussion by Beilinson** [pmndrs/postprocessing#473] [Q&A] [answered]
1. Why does setting <code>antialias: true</code> on the WebGLRenderer instance not run antialiasing on the EffectComposer? Could it not be applied on the final canvas either way?
2. What is the impact the following settings? How does the rendering change if either of them is enabled?
    <code>stencil: false, depth: false</code>
3. How might <code>preserveDrawingBuffer: true</code> affect the rendering, if at all?

**Comment by vanruesc** [comment, marked as answer]
MSAA supersamples the depth buffer. [This article](https://mynameismjp.wordpress.com/2012/10/24/msaa-overview/) provides a good overview of the technique.

1. `antialias: true` only applies to the canvas. The `RenderPass` in `postprocessing` renders geometry to an off-screen render target which may use MSAA via the [samples](https://threejs.org/docs/?q=rendert#api/en/renderers/WebGLRenderTarget.samples) setting. This setting is currently exposed through `EffectComposer.multisampling`. MSAA solely applies to the edges of triangles and fully relies on depth. Copying a processed image from a render target to the canvas is done by rendering a single triangle that covers the whole screen. MSAA won't be able to do anything in this case.
2. Depth is disabled on the canvas (`depth: false`) because it isn't needed for the blit operation, as explained above. The same applies to `antialias: false`: MSAA on the canvas would only waste space for unused higher resolution depth buffers.
The stencil buffer (`stencil: false`) is also not needed on the canvas because no stencil operations will be performed on the canvas. It all happens off-screen on render targets instead. The `EffectComposer` has a [stencilBuffer](https://github.com/pmndrs/postprocessing/blob/da28cd5d6800da9e6c10ffb4590a23258d7f766a/src/core/EffectComposer.js#L44) setting for that purpose.
4. I don't know. I think this setting is sometimes required if you want to copy the pixels of the canvas to an image. But other than that I think all it does is slow things down.
